Gateway/Validator/APMRequestValidator.php

```diff
@@ -81,6 +82,10 @@ private function validateAtomeAmount($order)
         $currency = strtoupper($order->getCurrencyCode());
         $amount = $order->getGrandTotalAmount();
 
+        if ($subTotal === 0.0) {
```

Comment:
Above, we cast `$price` to float. Is it not necessary here?

Reply:
`getGrandTotalAmount` return float value by default and I have also tested too.

```
  /**
   * Returns order grand total amount
   *
   * @return float
   */
  public function getGrandTotalAmount();
```

Reply:
let me cast to float for 100% sure.